src/components/composer/composer/abstract_composer_store.ts

```diff
@@ -601,37 +601,72 @@ export abstract class AbstractComposerStore extends SpreadsheetStore {
     return (token.parenthesesCode || "").startsWith(previousSymbolAtCursor.parenthesesCode || "");
   }
 
-  private computeParenthesisRelatedToCursor() {
-    if (
-      this.tokenAtCursor &&
-      ["LEFT_PAREN", "RIGHT_PAREN"].includes(this.tokenAtCursor.type) &&
-      this.tokenAtCursor.parenthesesCode !== ""
-    ) {
-      const matchingParenthesisIndexes = this.currentTokens.reduce((acc, t, index) => {
-        if (
-          ["LEFT_PAREN", "RIGHT_PAREN"].includes(t.type) &&
-          t.parenthesesCode === this.tokenAtCursor?.parenthesesCode
-        ) {
-          acc.push(index);
-        }
-        return acc;
-      }, [] as number[]);
+  private computeTokensRelatedToCursor() {
+    let matchingTokenIndexes: number[] = [];
 
-      if (matchingParenthesisIndexes.length >= 2) {
-        for (let i = 0; i < this.currentTokens.length; i++) {
-          this.currentTokens[i].isRelatedToCursorToken = matchingParenthesisIndexes.includes(i)
-            ? true
-            : false;
-        }
-        return;
+    if (this.tokenAtCursor) {
+      switch (this.tokenAtCursor.type) {
+        case "LEFT_PAREN":
+        case "RIGHT_PAREN":
+          if (this.tokenAtCursor.parenthesesCode) {
+            matchingTokenIndexes = this.currentTokens.reduce(
```

Comment:
a functionsimply looping on all the tokens would probably me easier to read than a  reduce

Reply:
I added a "[REF] to squash commit", tell me if you prefer this version